test_setuptools_version fails on my Macbook

`./pants test tests/python/pants_test/backend/python/:: -- -k test_setuptools_version`

Yields:

```
.pants.d/pyprep/sources/60babee1b95bc6d48c7194be264e52b39df17eda/pants_test/backend/python/tasks/test_setup_py.py:123: in assert_extra
                         self.assertEqual(expected_version, package.raw_version)
                     E   AssertionError: u'30.0.0' != '38.5.1'
```

Not sure where 38.5.1 is coming from - when the pants venv is created, it shows that 38.5.1 is found but then uninstalled, I assume in favor of 30.0.0. In any case, something about the test's interpreter is wrong.

--- a/tests/python/pants_test/backend/python/tasks/test_setup_py.py
+++ b/tests/python/pants_test/backend/python/tasks/test_setup_py.py
@@ -16,6 +16,7 @@ from twitter.common.collections import OrderedSet
 from twitter.common.dirutil.chroot import Chroot
 
 from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.backend.python.tasks.select_interpreter import SelectInterpreter
 from pants.backend.python.tasks.setup_py import SetupPy
 from pants.base.exceptions import TaskError
 from pants.build_graph.build_file_aliases import BuildFileAliases
@@ -43,7 +44,9 @@ class SetupPyTestBase(PythonTaskTestBase):
   @contextmanager
   def run_execute(self, target, recursive=False):
     self.set_options(recursive=recursive)
-    context = self.context(target_roots=[target])
+    si_task_type = self.synthesize_task_subtype(SelectInterpreter, 'si_scope')
+    context = self.context(for_task_types=[si_task_type], target_roots=[target])
+    si_task_type(context, os.path.join(self.pants_workdir, 'si')).execute()
     setup_py = self.create_task(context)
     setup_py.execute()
     yield context.products.get_data(SetupPy.PYTHON_DISTS_PRODUCT)

